src/units/abilities.cpp

```diff
@@ -1115,62 +1115,100 @@ unit_ability_list attack_type::get_weapon_ability(const std::string& ability) co
 	if(self_) {
 		abil_list.append((*self_).get_abilities(ability, self_loc_));
 		for(unit_ability_list::iterator i = abil_list.begin(); i != abil_list.end();) {
-			if(!((*i->ability_cfg)["overwrite_specials"].to_bool() && special_active(*i->ability_cfg, AFFECT_SELF, ability, true, "filter_student"))) {
+			if(!special_active(*i->ability_cfg, AFFECT_SELF, ability, true, "filter_student")) {
 				i = abil_list.erase(i);
 			} else {
 				++i;
 			}
 		}
-
-		if(abil_list.empty()){
-			abil_list.append((*self_).get_abilities(ability, self_loc_));
-			for(unit_ability_list::iterator i = abil_list.begin(); i != abil_list.end();) {
-				if(!special_active(*i->ability_cfg, AFFECT_SELF, ability, true, "filter_student")) {
-					i = abil_list.erase(i);
-				} else {
-					++i;
-				}
-			}
-		}
 	}
 
 	if(other_) {
 		abil_other_list.append((*other_).get_abilities(ability, other_loc_));
 		for(unit_ability_list::iterator i = abil_other_list.begin(); i != abil_other_list.end();) {
-			if(!((*i->ability_cfg)["overwrite_specials"].to_bool() && special_active_impl(other_attack_, shared_from_this(), *i->ability_cfg, AFFECT_OTHER, ability, true, "filter_student"))) {
+			if(!special_active_impl(other_attack_, shared_from_this(), *i->ability_cfg, AFFECT_OTHER, ability, true, "filter_student")) {
 				i = abil_other_list.erase(i);
 			} else {
 				++i;
 			}
 		}
-		if(abil_other_list.empty()){
-			abil_other_list.append((*other_).get_abilities(ability, other_loc_));
-			for(unit_ability_list::iterator i = abil_other_list.begin(); i != abil_other_list.end();) {
-				if(!special_active_impl(other_attack_, shared_from_this(), *i->ability_cfg, AFFECT_OTHER, ability, true, "filter_student")) {
-					i = abil_other_list.erase(i);
-				} else {
-					++i;
-				}
-			}
-		}
 	}
+
 	abil_list.append(abil_other_list);
+	if(!abil_list.empty()){
+		abil_list = overwrite_special_checking(ability, abil_list, abil_list, "filter_student");
+	}
 	return abil_list;
 }
 
 unit_ability_list attack_type::get_specials_and_abilities(const std::string& special) const
 {
 	unit_ability_list abil_list = get_weapon_ability(special);
-	for(unit_ability_list::iterator i = abil_list.begin(); i != abil_list.end();) {
-		if((*i->ability_cfg)["overwrite_specials"].to_bool()) {
-			return abil_list;
+	unit_ability_list spe_list = get_specials(special);
+	if(!spe_list.empty()){
+		spe_list = overwrite_special_checking(special, spe_list, abil_list);
+		if(special == "plague" && !spe_list.empty()){
+			return spe_list;
 		}
-		++i;
+		abil_list.append(spe_list);
 	}
-	abil_list.append(get_specials(special));
 	return abil_list;
 }
 
+static bool overwrite_special_affects(const config& special)
+{
+	const std::string& apply_to = special["overwrite_specials"];
+	if ( apply_to.empty() || apply_to == "none")
+		return false;
+	if ( apply_to == "one_side" || apply_to == "both_sides")
+		return true;
+	return false;
```

Comment:
I'd very much prefer this be just `return apply_to == "one_side" || apply_to == "both_sides"`.

Reply:
done